src/components/panel/panel.js

```diff
@@ -0,0 +1,284 @@
+ */
+angular
+    .module('material.components.panel', [
+      'material.core'
+    ])
+    .service('$mdPanel', MdPanelService);
+
+/**
+ * @ngdoc service
+ * @name $mdPanel
+ * @module material.components.panel
+ *
+ * @description
+ * `$mdPanel` is a general component that can be used for any panels on the screen,
+ * such as a tooltip, dialog, or custom popup.
+ *
+ * @usage
+ * <hljs lang="js">
+ * (function(angular, undefined) {
+ *   ‘use strict’;
+ *
+ *   angular
+ *       .module('demoApp', ['ngMaterial'])
+ *       .controller('PizzaBuilderController', PizzaBuilderController)
+ *       .controller('PizzaDialogController', DialogController);
+ *
+ *   function PizzaBuilderController($element, $mdPanel) {
+ *     var panelRef;
+ *     var toppings = [
+ *       'pepperoni',
+ *       'sausage',
+ *       'ham',
+ *       'peppers',
+ *       'onions',
+ *       'pineapple'
+ *     ];
+ *
+ *     function showPizzaPanel($event) {
+ *       var config = {
+ *         parent: angular.element(document.body),
+ *         controller: DialogController,
+ *         controllerAs: 'ctrl',
+ *         position: $element,
+ *         targetEvent: $event,
+ *         template:
+ *           '<div class="pizza-panel">' +
+ *           '  <h2>Choose your favorite topping.</h2>' +
+ *           '  <md-list>' +
+ *           '    <md-list-item ng-repeat="topping in ctrl.toppings">' +
+ *           '      <md-button ng-click="ctrl.favoriteTopping = topping">' +
+ *           '        {{ topping }}' +
+ *           '      </md-button>' +
+ *           '    </md-list-item>' +
+ *           '  </md-list>' +
+ *           '  <div class="actions">' +
+ *           '    <md-button ng-click="ctrl.closeDialog()">CANCEL</md-button>' +
+ *           '  </div>' +
+ *           '</div>',
+ *         clickOutsideToClose: true,
+ *         escapeToClose: true,
+ *         focusOnOpen: true
+ *       }
+ *       panelRef = $mdPanel.create(config);
+ *       panelRef.open()
+ *           .finally(function() {
+ *             panelRef = undefined;
+ *           });
+ *     }
+ *   }
+ *
+ *   function DialogController($mdPanelRef, toppings) {
+ *     var toppings;
+ *     var favoriteTopping;
+ *
+ *     function closeDialog() {
+ *       $mdPanelRef.close();
+ *     }
+ *   }
+ * })(angular);
+ * </hljs>
+ */
+
+/**
+ * @ngdoc method
+ * @name $mdPanel#create
+ * @description
+ * Creates a panel with the specified options.
+ *
+ * @param opt_config {Object=} Specific configuration object that may contain the
+ * following properties:
+ *
+ *   N/A
+ *
+ * TODO(ErinCoughlan): Add the following config options.
+ *   - `groupName` - `{string=}`: Name of panel groups. This group name is used for
+ *     configuring the number of open panels and identifying specific behaviors for
+ *     groups. For instance, all tooltips will be identified using the same groupName.
+ *   - `maxOpen` - `{number=}` : Maximum number of panel in a specific group that
```

Comment:
This works a lot better. There were a lot of edge cases with the option version, which I didn't like.

I added it to the design doc, but am leaving it out of this PR for now. We'll add it when we work on groups.